hw/mcu/ambiq/apollo3/src/hal_flash.c

```diff
@@ -168,6 +168,32 @@ apollo3_flash_write(const struct hal_flash *dev, uint32_t address,
     return rc;
 }
 
+static int
+apollo3_flash_write(const struct hal_flash *dev, uint32_t address,
+    const void *src, uint32_t num_bytes)
+{
+    int rc = 0;
+    uint32_t offset = 0;
+    uint32_t chunk_len;
+    uint8_t ram_buf[MYNEWT_VAL(FLASH_INTERMEDIARY_BUF_SIZE)];
```

Comment:
FLASH_INTERMEDIARY_BUF_SIZE is not in any syscfg.yml yet

Reply:
Added the definition to the mcu syscfg